radicle-node/src/wire/protocol.rs

```diff
@@ -0,0 +1,315 @@
+use std::collections::VecDeque;
+use std::io::{Read, Write};
+use std::net;
+use std::os::unix::io::AsRawFd;
+use std::os::unix::prelude::RawFd;
+use std::sync::Arc;
+use std::time::Duration;
+
+use cyphernet::addr::{Addr as _, HostAddr, PeerAddr};
+use cyphernet::crypto::Ecdh;
+use nakamoto_net::{DisconnectReason, Link, LocalTime};
```

Comment:
What would it take to drop the nakamoto dependency now? Is it just these three types? Could we put them in `netservices`?

Reply:
1. I need to understand better the way `LocalTime` works, maybe it deserves dedicated crate or can be moved into `amplify` if it useful in multiple projects.

2. I have something like `Link` in the `netservices`, at least I was doing it. If it is still there, we can switch on it.

3. Disconnect reason is strange thing for me, since it appeared I was able to deal w/o it in the new reactor and netservices, so potentially we can avoid it here as well.